File: hw08/src/main/java/ru/otus/hw/models/Book.java
package ru.otus.hw.models;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@EqualsAndHashCode(exclude = {"author", "genres"})
@ToString(exclude = {"author", "genres"})
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "books")
public class Book {

    @Id
    private String id;

    private String title;

    @DBRef(lazy = true)
    private Author author;

    @DBRef(lazy = true)
    private List<Genre> genres;

}

File: hw10-client/src/app/model/book.ts
import { Author } from './author';
import { Genre } from './genre';

export class Book {
  id: number;
  title: string;
  author: Author;
  genres: Genre[];

  constructor(id: number, title: string, author: Author, genres: Genre[]){
    this.id = id;
    this.title = title;
    this.author = author;
    this.genres = genres;
    }
}
